File: src/main/java/io/tbbc/cf/turret/chassis/skin/ChassisSkin.java
package io.tbbc.cf.turret.chassis.skin;

import io.tbbc.cf.research.Research;

import java.util.List;

public record ChassisSkin(String name, String label, ChassisEgoSkinProps egoSkinProps,
                          List<Research> requiredResearch) {
}

File: src/main/webui/src/model/turret/ChassisSkin.ts
import Research from "../common/Research";

export default class ChassisSkin {
    name: string;
    label: string;
    icon: string;
    requiredResearch: Research[];
    constructor(name: string, label: string, icon: string, requiredResearch: Research[]) {
        this.name = name;
        this.label = label;
        this.icon = icon;
        this.requiredResearch = requiredResearch;
    }
}